File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraVerticalGrid;
using DevExpress.Utils;
using DevExpress.XtraBars;
using System.Reflection;

namespace Sample___Toolbar {
	public partial class Form1 : Form {
		public Form1() {
			InitializeComponent();
			this.barButtonCategorized.DownChanged += this.barButton_DownChanged;
			this.barButtonAlphabetical.DownChanged += this.barButton_DownChanged;
			SetBarButtonToolTip(this.barButtonAlphabetical, "Alphabetical");
			SetBarButtonToolTip(this.barButtonCategorized, "Categorized");
			this.barButtonCategorized.Glyph = ResourceImageHelper.CreateImageFromResources("Sample___Toolbar.Categorized.png", Assembly.GetExecutingAssembly());
            this.barButtonAlphabetical.Glyph = ResourceImageHelper.CreateImageFromResources("Sample___Toolbar.Alphabetical.png", Assembly.GetExecutingAssembly());
			this.ShowCategories = true;
			PropertyGridControl.SelectedObject = PropertyGridControl;
		}
		public bool ShowCategories {
            get { return this.barButtonCategorized.Down; }
            set { this.barButtonCategorized.Down = value; }
        }
        public PropertyGridControl PropertyGridControl {
            get { return propertyGridControl; }
        }
        private void barButton_DownChanged(object sender, ItemClickEventArgs e) {
			if(e.Item.Equals(this.barButtonCategorized)) {
				SetBarButtonDown(this.barButtonAlphabetical, !this.barButtonCategorized.Down);
			} else {
				SetBarButtonDown(this.barButtonCategorized, !this.barButtonAlphabetical.Down);
			}
            UpdatePropertyGrid();
        }
        static void SetBarButtonToolTip(DevExpress.XtraBars.BarButtonItem barButton, string value) {
            SuperToolTip superToolTip = new SuperToolTip();
            ToolTipTitleItem toolTipTitleItem = new ToolTipTitleItem();
            toolTipTitleItem.Text = value;
            superToolTip.Items.Add(toolTipTitleItem);
            barButton.SuperTip = superToolTip;
        }
        void SetBarButtonDown(BarButtonItem barButton, bool value) {
            barButton.DownChanged -= new ItemClickEventHandler(this.barButton_DownChanged);
            barButton.Down = value;
            barButton.DownChanged += new ItemClickEventHandler(this.barButton_DownChanged);
        }
        void UpdatePropertyGrid() {
            this.propertyGridControl.OptionsView.ShowRootCategories = this.barButtonCategorized.Down;
        }
	}
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraVerticalGrid
Imports DevExpress.Utils
Imports DevExpress.XtraBars
Imports System.Reflection

Namespace Sample___Toolbar
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			AddHandler Me.barButtonCategorized.DownChanged, AddressOf barButton_DownChanged
			AddHandler Me.barButtonAlphabetical.DownChanged, AddressOf barButton_DownChanged
			SetBarButtonToolTip(Me.barButtonAlphabetical, "Alphabetical")
			SetBarButtonToolTip(Me.barButtonCategorized, "Categorized")
			Me.barButtonCategorized.Glyph = ResourceImageHelper.CreateImageFromResources("Sample___Toolbar.Categorized.png", System.Reflection.Assembly.GetExecutingAssembly())
			Me.barButtonAlphabetical.Glyph = ResourceImageHelper.CreateImageFromResources("Sample___Toolbar.Alphabetical.png", System.Reflection.Assembly.GetExecutingAssembly())
			Me.ShowCategories = True
			PropertyGridControl.SelectedObject = PropertyGridControl
		End Sub
		Public Property ShowCategories() As Boolean
			Get
				Return Me.barButtonCategorized.Down
			End Get
			Set(ByVal value As Boolean)
				Me.barButtonCategorized.Down = value
			End Set
		End Property
		Public ReadOnly Property PropertyGridControl() As PropertyGridControl
			Get
				Return propertyGridControl_Renamed
			End Get
		End Property
		Private Sub barButton_DownChanged(ByVal sender As Object, ByVal e As ItemClickEventArgs)
			If e.Item.Equals(Me.barButtonCategorized) Then
				SetBarButtonDown(Me.barButtonAlphabetical, (Not Me.barButtonCategorized.Down))
			Else
				SetBarButtonDown(Me.barButtonCategorized, (Not Me.barButtonAlphabetical.Down))
			End If
			UpdatePropertyGrid()
		End Sub
		Private Shared Sub SetBarButtonToolTip(ByVal barButton As DevExpress.XtraBars.BarButtonItem, ByVal value As String)
			Dim superToolTip As New SuperToolTip()
			Dim toolTipTitleItem As New ToolTipTitleItem()
			toolTipTitleItem.Text = value
			superToolTip.Items.Add(toolTipTitleItem)
			barButton.SuperTip = superToolTip
		End Sub
		Private Sub SetBarButtonDown(ByVal barButton As BarButtonItem, ByVal value As Boolean)
			RemoveHandler barButton.DownChanged, AddressOf barButton_DownChanged
			barButton.Down = value
			AddHandler barButton.DownChanged, AddressOf barButton_DownChanged
		End Sub
		Private Sub UpdatePropertyGrid()
			Me.propertyGridControl_Renamed.OptionsView.ShowRootCategories = Me.barButtonCategorized.Down
		End Sub
	End Class
End Namespace